pkg/fake/orchestrator/rest/fake_orchestrator_rest_server.go

```diff
@@ -170,28 +170,53 @@ func (s *FakeOrchestratorRESTServer) SetupFakeOrchestratorRESTServer() string {
 				return
 			}
 			return
-		// Create Resources (POST)
-		case urlMatches(path, orchestrator.CreateOrAttachResourcePOSTURL) && r.Method == http.MethodPost:
+		// Create Resources (PUT)
+		case urlMatches(path, orchestrator.CreateResourcePUTURL) && r.Method == http.MethodPut:
 			resource := &paragliderpb.ResourceDescriptionString{}
 			err := json.Unmarshal(body, resource)
 			if err != nil {
 				http.Error(w, fmt.Sprintf("error unmarshalling request body: %s", err), http.StatusBadRequest)
 			}
-			err = s.writeResponse(w, map[string]string{"name": resource.Name})
+			err = s.writeResponse(w, &paragliderpb.CreateResourceResponse{Name: strings.Split(path, "/")[len(strings.Split(path, "/"))-1], Uri: "testUri", Ip: "testIp"})
 			if err != nil {
 				http.Error(w, fmt.Sprintf("error writing response: %s", err), http.StatusInternalServerError)
 			}
 			return
-		// Create Resources (PUT)
-		case urlMatches(path, orchestrator.CreateResourcePUTURL) && r.Method == http.MethodPut:
-			resource := &paragliderpb.ResourceDescriptionString{}
-			err := json.Unmarshal(body, resource)
+		// Create or Attach Resources (POST)
+		case urlMatches(path, orchestrator.CreateOrAttachResourcePOSTURL) && r.Method == http.MethodPost:
+			var bodyMap map[string]interface{}
+			err := json.Unmarshal(body, &bodyMap)
 			if err != nil {
 				http.Error(w, fmt.Sprintf("error unmarshalling request body: %s", err), http.StatusBadRequest)
+				return
 			}
-			err = s.writeResponse(w, map[string]string{"name": strings.Split(path, "/")[len(strings.Split(path, "/"))-1]})
-			if err != nil {
-				http.Error(w, fmt.Sprintf("error writing response: %s", err), http.StatusInternalServerError)
+
+			// Create and Attach share the same URL with POST.
+			// If the body has an "id" field, it is an Attach request. Otherwise, Create request.
+			if bodyMap["id"] == nil {
+				// Create Resource (POST)
+				resource := &paragliderpb.ResourceDescriptionString{}
+				err = json.Unmarshal(body, resource)
+				if err != nil {
+					http.Error(w, fmt.Sprintf("error unmarshalling request body: %s", err), http.StatusBadRequest)
+				}
+
+				err = s.writeResponse(w, &paragliderpb.CreateResourceResponse{Name: resource.Name, Uri: "testUri", Ip: "testIp"})
+				if err != nil {
+					http.Error(w, fmt.Sprintf("error writing response: %s", err), http.StatusInternalServerError)
+				}
+			} else {
+				// Attach Resource (POST)
+				resource := &paragliderpb.ResourceString{}
+				err := json.Unmarshal(body, resource)
+				if err != nil {
+					http.Error(w, fmt.Sprintf("error unmarshalling request body: %s", err), http.StatusBadRequest)
+				}
+
+				err = s.writeResponse(w, &paragliderpb.AttachResourceResponse{Name: "validResourceName", Uri: resource.Id, Ip: "testIp"})
```

Comment:
Nit: make these constants instead of having them inline?

Reply:
Done